src/components/responsive-table/table.tsx

```diff
@@ -77,6 +77,21 @@ class Table extends React.Component<
   }
 
   componentDidMount(): void {
+    this.updateTableWidth();
+  }
+
+  componentDidUpdate(_, prevState): void {
+    const { rows } = this.state;
+    if (prevState.rows !== rows) {
```

Comment:
what the type of rows ? if there a depth structure as
`const rows = {
 columns: {
  someField: 'kek'
 },
 a: 'a'
}`
I would use deepEqual method

Reply:
you can write by yourself that method, but there might popup unexpected things or use that library 
https://www.npmjs.com/package/deep-equal